api/test/integration/cost-surface-repo/world.ts

```diff
@@ -0,0 +1,80 @@
+import { INestApplication } from '@nestjs/common';
+import { ScenariosPlanningUnitGeoEntity } from '../../../src/modules/scenarios-planning-unit/entities/scenarios-planning-unit.geo.entity';
+import { GivenScenarioPuDataExists } from '../../steps/given-scenario-pu-data-exists';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { ScenariosPuCostDataGeo } from '../../../src/modules/analysis/providers/cost-surface/adapters/scenarios-pu-cost-data.geo.entity';
+import { DbConnections } from '../../../src/ormconfig.connections';
+import { In, Repository } from 'typeorm';
+import { v4 } from 'uuid';
+
+export interface CostSurfaceUpdateWorld {
+  cleanup: () => Promise<void>;
+  scenarioId: string;
+  planningUnitsIds: string[];
+  GivenPuCostDataExists: () => Promise<string[]>;
+  GetPuCostsData: (
+    scenarioId: string,
+  ) => Promise<{ scenario_id: string; cost: number; pu_id: string }[]>;
+}
+
+export const createWorld = async (
+  app: INestApplication,
+): Promise<CostSurfaceUpdateWorld> => {
+  const scenarioId = v4();
+  const puCostRepoToken = getRepositoryToken(
+    ScenariosPuCostDataGeo,
+    DbConnections.geoprocessingDB,
+  );
+  const puDataRepoToken = getRepositoryToken(
+    ScenariosPlanningUnitGeoEntity,
+    DbConnections.geoprocessingDB,
+  );
+  const puDataRepo: Repository<ScenariosPlanningUnitGeoEntity> = app.get(
+    puDataRepoToken,
+  );
+  const puCostDataRepo: Repository<ScenariosPuCostDataGeo> = app.get(
+    puCostRepoToken,
+  );
+  const scenarioPuData = await GivenScenarioPuDataExists(
+    puDataRepo,
+    scenarioId,
+  );
+
+  const puDataIds = scenarioPuData.rows.map((row) => row.id);
+  const puIds = scenarioPuData.rows.map((row) => row.puGeometryId);
+
+  return {
+    GetPuCostsData: async (
+      scenarioId: string,
+    ): Promise<{ scenario_id: string; cost: number; pu_id: string }[]> =>
+      puCostDataRepo.query(`
+      select spud.scenario_id, spucd."cost", spucd.output_results_data_id as pu_id from scenarios_pu_data as spud join scenarios_pu_cost_data as spucd on (spud."id" = spucd.scenarios_pu_data_id)
+      where spud.scenario_id = '${scenarioId}'
+`),
```

Comment:
what is the reason behind using a raw query?

Reply:
For a sake of simplicity and missing relations :(